### PR TITLE
Feature/logging

### PR DESCRIPTION
**Pull request by tatianaIljina**

### CODE REVIEW

File: RTS/Tenders.RTS.Bidding/Program.cs [hunk 1]
```diff
@@ -47,6 +47,7 @@ public static async Task Main(string[] args)
 
         private static void ConfigureServices(HostBuilderContext context, IServiceCollection sc)
         {
+            sc.AddNlogLogging(context.Configuration);
```
Comment (DamienVStone) [inline RTS/Tenders.RTS.Bidding/Program.cs:50]: look at line 55

File: RTS/Tenders.RTS.Bidding/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -3,8 +3,10 @@
...
 using System;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 
 #pragma warning disable
 [assembly: CLSCompliant(false)]
-[assembly: ComVisible(false)]
\ No newline at end of file
+[assembly: ComVisible(false)]
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2")]
```
Comment (DamienVStone) [inline RTS/Tenders.RTS.Bidding/Properties/AssemblyInfo.cs:12]: move it to .csproj file

File: RTS/Tenders.RTS.Bidding/Services/CometClient.cs [hunk 3]
```diff
@@ -169,7 +156,7 @@ private async Task<ICometData[]> HandshakeAsync(CancellationToken ct, WebProxy p
             {
                 this.SetState(State.Handshaking, State.New);
                 var ex = new InvalidOperationException("Cannot handshake.");
-                this.logger.Error(ex);
+                this.logger.LogError(ex.Message);
```
Comment (DamienVStone) [inline RTS/Tenders.RTS.Bidding/Services/CometClient.cs:159]: use (Exception, string) overload

File: RTS/Tenders.RTS.Bidding/Services/CometClient.cs [hunk 4]
```diff
@@ -187,7 +174,7 @@ private async Task<ICometData[]> SubscribeAsync(ICometData[] data, string auctio
             {
                 this.SetState(State.Subscribing, State.New);
                 var ex = new InvalidOperationException("Cannot subscribe.");
-                this.logger.Error(ex);
+                this.logger.LogError(ex.Message);
```
Comment (DamienVStone) [inline RTS/Tenders.RTS.Bidding/Services/CometClient.cs:177]: and here

File: Services/Addressees/Addressees.API/Models/UniqueIndexViolationException.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+namespace Addressees.API.Models
+{
+    using System;
+
+#pragma warning disable S3925 // "ISerializable" should be implemented correctly
+    /// <summary>
+    /// Unique Index Violation Exception.
+    /// </summary>
+    /// <seealso cref="System.Exception" />
+    public class UniqueIndexViolationException : Exception
```
Comment (DamienVStone) [inline Services/Addressees/Addressees.API/Models/UniqueIndexViolationException.cs:14]: Is it logging part too?

File: tests/Tenders.Core.Tests/Tenders.Core.Tests.csproj [hunk 6]
```diff
@@ -57,6 +57,7 @@
   <ItemGroup>
     <ProjectReference Include="..\..\API\Tenders.API\Tenders.API.csproj" />
     <ProjectReference Include="..\..\Core\Tenders.Core.EPPlusAdapter\Tenders.Core.EPPlusAdapter.csproj" />
+    <ProjectReference Include="..\..\Core\Tenders.Core.NLogLoggerService\Tenders.Core.NLogLoggerService.csproj" />
```
Comment (DamienVStone) [inline tests/Tenders.Core.Tests/Tenders.Core.Tests.csproj:60]: Is it need?

**Review by DamienVStone — CHANGES_REQUESTED** [review]

File: tests/Tenders.Core.Tests/Tenders.Core.Tests.csproj (hunk 6 above)
Comment (tatianaIljina) [reply tests/Tenders.Core.Tests/Tenders.Core.Tests.csproj:60]: yes, for LoggerServiceTest class

File: Services/Addressees/Addressees.API/Models/UniqueIndexViolationException.cs (hunk 5 above)
Comment (tatianaIljina) [reply Services/Addressees/Addressees.API/Models/UniqueIndexViolationException.cs:14]: no, of course. However, I saw unnecessary reference and decided to fix it as a part of this small pr :)

**Review by Anny777 — APPROVED** [review]

**Review by DamienVStone — APPROVED** [review]

**Pull request merged**